src/helpers/completion.ts

```diff
@@ -187,6 +188,19 @@ export const readData =
   ) =>
   (writer: (data: string) => void): Promise<string> =>
     new Promise(async (resolve) => {
+      const rl = readline.createInterface({
+        input: process.stdin,
+      });
+
+      rl.input.setRawMode(true);
+
+      rl.input.on('keypress', (key) => {
+        if (typeof key === 'string' && key.toLowerCase().includes('q')) {
```

Comment:
this is a great idea. can/should we support the `esc` key as well? or is that not possible. i'm guessing most users will intuitively hit `esc` when wanting to stop the stream, so if that works as well it could be a nice addition